extensions/wikia/DefaultMessages/DefaultMessagesCache.php

```diff
@@ -0,0 +1,521 @@
+	 * $wgMaxMsgCacheEntrySize are assigned a special value, and are loaded
+	 * on-demand from the database later.
+	 *
+	 * @param $code Optional language code, see documenation of load().
+	 * @return Array: Loaded messages for storing in caches.
+	 */
+	function loadFromDB( $code = false ) {
+		wfProfileIn( __METHOD__ );
+		global $wgMaxMsgCacheEntrySize;
+		global $wgDefaultMessagesDB;
+		$dbr = wfGetDB( DB_SLAVE, array(), $wgDefaultMessagesDB );
+		$cache = array();
+
+		# Common conditions
+		$conds = array(
+			'page_is_redirect' => 0,
+			'page_namespace' => NS_MEDIAWIKI,
+		);
+
+		if ( $code ) {
+			# Is this fast enough. Should not matter if the filtering is done in the
+			# database or in code.
+			if ( $code !== 'en' ) {
+				# Messages for particular language
+				$conds[] = "page_title" . $dbr->buildLike( $dbr->anyString(), "/$code" );
+			} else {
+				# Effectively disallows use of '/' character in NS_MEDIAWIKI for uses
+				# other than language code.
+				$conds[] = "(page_title not like '%%/%%') OR (page_title like '%%/en')";
+			}
+		}
+
+		# Conditions to fetch oversized pages to ignore them
+		$bigConds = $conds;
+		$bigConds[] = 'page_len > ' . intval( $wgMaxMsgCacheEntrySize );
+
+		# Load titles for all oversized pages in the MediaWiki namespace
+		$res = $dbr->select( "page", 'page_title', $bigConds, __METHOD__ );
+		while ( $row = $dbr->fetchObject( $res ) ) {
+			$cache[$row->page_title] = '!TOO BIG';
+		}
+		$dbr->freeResult( $res );
+
+		# Conditions to load the remaining pages with their contents
+		$smallConds = $conds;
+		$smallConds[] = 'page_latest=rev_id';
+		$smallConds[] = 'rev_text_id=old_id';
+		$smallConds[] = 'page_len <= ' . intval( $wgMaxMsgCacheEntrySize );
+
+		$res = $dbr->select( array( "page", "revision", "text" ),
+			array( 'page_title', 'old_text', 'old_flags' ),
+			$smallConds, __METHOD__ );
+
+		while ( $row = $dbr->fetchObject( $res ) ) {
+			$cache[$row->page_title] = ' ' . Revision::getRevisionText( $row );
+		}
+		$dbr->freeResult( $res );
+
+		$cache['VERSION'] = MSG_CACHE_VERSION;
+		wfProfileOut( __METHOD__ );
+		return $cache;
+	}
+
+	/**
+	 * Shortcut to update caches.
+	 *
+	 * @param $cache Array: cached messages with a version.
+	 * @param $cacheKey String: Identifier for the cache.
+	 * @param $memc Bool: Wether to update or not memcache.
+	 * @param $code String: Language code.
+	 * @return False on somekind of error.
+	 */
+	protected function saveToCaches( $cache, $memc = true, $code = false ) {
+		wfProfileIn( __METHOD__ );
+
+		if ( F::app()->wg->AllowMemcacheDisable && ( F::app()->wg->AllowMemcacheWrites == false ) ) {
+			return true;
+		}
+
+		$cacheKey = $this->memcKey( 'messages', $code );
+		$statusKey = $this->memcKey( 'messages', $code, 'status' );
+
+		$success = $this->mMemc->add( $statusKey, 'loading', MSG_LOAD_TIMEOUT );
+		if ( !$success ) {
+			wfProfileOut( __METHOD__ );
+			return true; # Other process should be updating them now
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Move this trailing comment on the previous empty line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS139)